pyomo/common/tempfiles.py

```diff
@@ -428,6 +429,8 @@ def _resolve_tempdir(self, dir=None):
             return self.tempdir
         elif self.manager().tempdir is not None:
             return self.manager().tempdir
+        elif TempfileManager is not None and TempfileManager.main.tempdir is not None:
```

Comment:
I don't think you need the test for `TempfileManager is not None`, do you?

Reply:
You are right, I'm removing that check now